src/NuGet.Core/NuGet.ProjectManagement/Projects/BuildIntegratedNuGetProject.cs

```diff
@@ -307,19 +307,20 @@ public IReadOnlyList<PackageSpec> GetPackageSpecsForRestore(ExternalProjectRefer
             {
                 foreach (var reference in references)
                 {
-                    MSBuildRestoreUtility.AddMSBuildProjectReference(
-                        packageSpec,
-                        new ProjectRestoreReference
-                        {
-                            ProjectUniqueName = reference.UniqueName,
-                            ProjectPath = reference.MSBuildProjectPath
-                        },
-                        new LibraryDependency
-                        {
-                            LibraryRange = new LibraryRange(
-                                reference.UniqueName,
-                                LibraryDependencyTarget.ExternalProject)
-                        });
+                    // This reference applies to all frameworks
+                    // Include/exclude flags may be applied later when merged with project.json
+                    var projectReference = new ProjectRestoreReference
+                    {
+                        ProjectUniqueName = reference.UniqueName,
+                        ProjectPath = reference.MSBuildProjectPath
+                    };
+
+                    foreach (var framework in packageSpec.TargetFrameworks.Select(e => e.FrameworkName))
+                    {
+                        var group = new ProjectRestoreMetadataFrameworkInfo(framework);
```

Comment:
>ProjectRestoreMetadataFrameworkInfo [](start = 40, length = 35)

This will be created (framework count) x (project reference count) times, right? Needn't we only initialize this group once per framework?